.github/workflows/test_sanity_check.yml

```diff
@@ -0,0 +1,86 @@
+name: TheRock Sanity Check
+
+on:
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+  push:
+    branches:
+      - ADHOCBUILD
+
+jobs:
+  test_sanity_check:
+    name: "Sanity ROCM Test"
+    runs-on: linux-mi300-1gpu-ossci-rocm
+    strategy:
+      fail-fast: false
+    defaults:
+      run:
+        shell: bash
+    env:
+      VENV_DIR: ${{ github.workspace }}/.venv
+      ARTIFACT_RUN_ID: "${{ inputs.artifact_run_id != '' && inputs.artifact_run_id || github.run_id }}"
+      BUILD_ARTIFACTS_DIR: "build/artifacts"
+      BIN_DIR: "${{ github.workspace }}/build/artifacts/output_dir/bin"
```

Comment:
I would namespace this environment variable `THE_ROCK_BIN_DIR` or `THEROCK_BIN_DIR` (check if we use either spacing in other parts of the repo). Generic and short variables names could be already set on the system or set by one of the subprojects